tests/cli/test_tree.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+import sys
+import os
+from shutil import copyfile
+
+# Following https://stackoverflow.com/questions/13493288/python-cli-program-unit-testing/13500346#13500346
+# and https://docs.pytest.org/en/latest/reference.html#testdir
+
+pytest_plugins = "pytester"
+
+@pytest.fixture
+def run(testdir):
+    def do_run(*args):
+        args = ["augur", "tree"] + list(args)
+        return testdir.run(*args)
+    return do_run
```

Comment:
This is really interesting. I haven't seen `pytester` used in this way. Clever! Usually pytester is used to test pytest plugins, but this is pretty cool. 

Have you seen [pytest-console-scripts](https://github.com/kvas-it/pytest-console-scripts)? I've used it in a few places around the Jupyter stack. Though, I'm not sure how it works with output *files* that you want to read and assert its contents.

Reply:
This was _literally_ the first thing I've done in pytest, so it was a steep learning curve!

Reply:
pytest is super clean but can be difficult to follow. Fixtures are often black boxes!